include/ginkgo/core/solver/chebyshev.hpp

```diff
@@ -0,0 +1,245 @@
+ * solution_{i-1})
+ * ```
+ *
+ *
+ * @tparam ValueType  precision of matrix elements
+ *
+ * @ingroup solvers
+ * @ingroup LinOp
+ */
+template <typename ValueType = default_precision>
+class Chebyshev
+    : public EnableLinOp<Chebyshev<ValueType>>,
+      public EnablePreconditionedIterativeSolver<ValueType,
+                                                 Chebyshev<ValueType>>,
+      public EnableApplyWithInitialGuess<Chebyshev<ValueType>>,
+      public Transposable {
+    friend class EnableLinOp<Chebyshev>;
+    friend class EnablePolymorphicObject<Chebyshev, LinOp>;
+    friend class EnableApplyWithInitialGuess<Chebyshev>;
+
+public:
+    using value_type = ValueType;
+    using transposed_type = Chebyshev<ValueType>;
+
+    std::unique_ptr<LinOp> transpose() const override;
+
+    std::unique_ptr<LinOp> conj_transpose() const override;
+
+    /**
+     * Return true as iterative solvers use the data in x as an initial guess.
+     *
+     * @return true as iterative solvers use the data in x as an initial guess.
+     */
+    bool apply_uses_initial_guess() const override
+    {
+        return this->get_default_initial_guess() ==
+               initial_guess_mode::provided;
+    }
+
+    /**
+     * Copy-assigns a Chebyshev solver. Preserves the executor, shallow-copies
+     * inner solver, stopping criterion and system matrix. If the executors
+     * mismatch, clones inner solver, stopping criterion and system matrix onto
+     * this executor.
+     */
+    Chebyshev& operator=(const Chebyshev&);
+
+    /**
+     * Move-assigns a Chebyshev solver. Preserves the executor, moves inner
+     * solver, stopping criterion and system matrix. If the executors mismatch,
+     * clones inner solver, stopping criterion and system matrix onto this
+     * executor. The moved-from object is empty (0x0 and nullptr inner solver,
+     * stopping criterion and system matrix)
+     */
+    Chebyshev& operator=(Chebyshev&&);
+
+    /**
+     * Copy-constructs an Chebyshev solver. Inherits the executor,
+     * shallow-copies inner solver, stopping criterion and system matrix.
+     */
+    Chebyshev(const Chebyshev&);
+
+    /**
+     * Move-constructs an Chebyshev solver. Preserves the executor, moves inner
+     * solver, stopping criterion and system matrix. The moved-from object is
+     * empty (0x0 and nullptr inner solver, stopping criterion and system
+     * matrix)
+     */
+    Chebyshev(Chebyshev&&);
+
+    class Factory;
+
+    struct parameters_type
+        : enable_preconditioned_iterative_solver_factory_parameters<
+              parameters_type, Factory> {
+        /**
+         * The pair of foci of ellipse, which covers the eigenvalues of
+         * preconditioned system. It is usually be {lower bound of eigval, upper
+         * bound of eigval} of preconditioned real matrices.
+         */
+        std::pair<value_type, value_type> GKO_FACTORY_PARAMETER_VECTOR(
```

Comment:
Is there a reason to use the old-style parameters, instead of the deferred ones?

Reply:
I think deferred factory, which handles the Factory with/without on, only for those Factory which requires `.on(exec)`. For the other value or pure LinOp parameter, we still use the original style.